src/graph/graphQLClient.ts

```diff
@@ -0,0 +1,86 @@
+import { GraphQLClient } from 'graphql-request';
+
+import {
+  GetSavedItemsDocument,
+  GetSavedItemsQuery,
+  GetSavedItemsQueryVariables,
+  SaveArchiveDocument,
+  SaveArchiveMutation,
+  SaveFavoriteDocument,
+  SaveFavoriteMutation,
+  SaveFavoriteMutationVariables,
+} from '../generated/graphql/types';
+import { SaveArchiveMutationVariables } from '../generated/graphql/types';
+import config from '../config';
+
+/**
+ * gives a graphQLClient for pocket-graph url
+ *
+ * This client initializes a `graphql-request` client
+ * @param access_token accessToken of the user
+ * @param consumer_key consumerKey assocuated with the user
+ */
+export function getClient(accessToken: string, consumerKey: string) {
+  return new GraphQLClient(
+    `${config.graphQLProxy}?consumer_key=${consumerKey}&access_token=${accessToken}`,
+    {
+      //fetch implementation used by node version,
+      //can give custom fetch package
+      fetch,
+    }
+  );
+}
+/**
+ * Calls saveArchive mutation
+ *
+ * @param auth auth headers from the web repo
+ * @param variables variables required for the mutation
+ */
+export async function callSaveArchive(
+  auth: any,
```

Comment:
Wondering if we can use `string` type for `auth`? I think headers are sent as a string over a request

Reply:
ah that's a copy-paste mistake! thanks for capturing this, will remove `auth` details for now - we are scoping to get accessToken and consumerKey from web repo as header

we have a follow-up investigation ticket to capture all auth patterns as we redirect from dotcom gateway...but when web repo sends the request to proxy, it can send accessToken/consumerKey as header

Reply:
✅